### ISSUE TITLE
add discriminative loss for mlp and y-hat

### ISSUE THREAD

**Issue by dribnet** [vdumoulin/discgen#5]
This code has been working great for me. Currently, the discriminative regularization in this codebase uses the loss from the batch norm layers of the ConvolutionalSequence, but the paper covers the more general case of taking loss from elsewhere in the classifier, as shown below:

![4654ee39-75d4-463c-b808-8bd4a2c5e616](https://cloud.githubusercontent.com/assets/945979/16356358/b657f77c-3b28-11e6-8acc-aacbb7dbc7cf.png)

I'd like to add losses at the higher layers - MLP and y-hat. I'm curious if anyone else is interested in working on this with me - the discriminative regularization parts are some of trickier parts of building the training computation graph, and y-hat currently doesn't have a named node in the classifier graph.

**Comment by vdumoulin** [comment]
I think that would be a good idea. I won't have time to work on it, but if you have questions or would like guidance I can sporadically chime in (and of course review the PR).
